### ISSUE TITLE
Ansible not enabling pipelining despite attempting to enable it three different ways

### ISSUE THREAD

**Issue by kevinburkerengin** [ansible/ansible#15083]
##### Issue Type:
- Bug Report
##### Ansible Version:

```
ansible 2.0.1.0
  config file = /Users/kevin/code/web-deployment/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

Here's my ansible.cfg

```
[defaults]
retry_files_enabled = False
roles_path = roles

[ssh_connection]
pipelining = True
control_path = /tmp/ansible-ssh-%%h-%%p-%%r
```
##### Environment:

Running on a Mac, attempting to deploy to a Ubuntu 15.10 instance.
##### Summary:

I'm trying to get Ansible's pipelining feature to work, since currently a playbook with 130 plays takes about 5 minutes and 30 seconds for me. Unfortunately it doesn't appear to be working; when I run ansible-playbook with the `-vvv` flag, every command is being prefaced with `ESTABLISH CONNECTION … EXEC ssh… PUT … EXEC SSH ….` The shortest possible play takes about 1.5 seconds to execute, which makes a playbook with 129 tasks take 5 minutes and 30 seconds.
##### Steps To Reproduce:

I've tried putting

```
[ssh_connection]
pipelining=True 
```

in ansible.cfg, changing my running command to ansible-playbook --extra-vars="pipelining=True", and changing the running command to `ANSIBLE_SSH_PIPELINING=1 ansible-playbook ...` None of these changed the running commands in `-vvv` mode, or the overall runtime. 

My project is available here: https://github.com/kevinburke/web-deployment. I usually run `make website`.
##### Expected Results:

I expected Ansible to run only one `EXEC` command for each task, instead of `EXEC... PUT... EXEC`. I expected the overall runtime to be shorter than 5 minutes and 30 seconds to run ~130 shell commands.
##### Actual Results:

Ansible continued to run without pipelining mode.

I read somewhere that `requiretty` in `/etc/sudoers` makes Ansible disable pipelining; I verified the remote host does not have this. I'm running Ansible on a Mac and connecting to a single Ubuntu 15.10 instance. I don't have anything funky in ssh_args, either.

I understand the team is busy but it seems like this feature is pretty important to having a good experience with Ansible, and I tried my best to read the docs, ask in the IRC channel, and debug this myself, and I'm at loose ends. I'm hoping that the results of this can/will turn into better documentation.

**Comment by seenickcode** [comment]
@kevinburkerengin was there ever a solution to this?

**Comment by andrewchen5678** [comment]
Same here. Need to switch to something else because ansible is very slow and the pipelining setting seem to be ignored.

**Comment by luqasz** [comment]
In ansible 2.4 I had to add 
```
[connection]
pipelining = True
```
Which is not in section `ssh_connection`

**Comment by bcoca** [comment]
this was a bug in 2.4, fixed in  #31244 which will be part of 2.4.1
